airflow/providers/elasticsearch/log/es_task_handler.py

```diff
@@ -98,6 +98,15 @@ def __init__(
         log_id_template: str | None = None,
     ):
         es_kwargs = es_kwargs or {}
+        # For elasticsearch>8,arguments like retry_timeout have changed for elasticsearch to retry_on_timeout
+        # in Elasticsearch() compared to previous versions.
+        # Read more at: https://elasticsearch-py.readthedocs.io/en/v8.8.2/api.html#module-elasticsearch
```

Comment:
if es_kwargs.get("retry_timeout"):
    es_kwargs["retry_on_timeout"] = es_kwargs.pop("retry_timeout")

I think the logic should be like this

Reply:
changed it: https://github.com/apache/airflow/pull/33281/commits/2984194c7f3be440831c6b92d577271c9027600b